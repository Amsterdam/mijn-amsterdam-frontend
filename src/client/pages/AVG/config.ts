import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
