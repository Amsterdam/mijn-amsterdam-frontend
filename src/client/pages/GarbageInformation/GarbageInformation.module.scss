@import '../../styles/_global.scss';

@mixin GarbagePanel {
  margin: 0 auto;
  padding: 0;
  max-width: 64rem;

  @include mq-tablet {
    margin-left: 9.5rem;
  }

  @include mq-phone {
    margin-left: 0;
  }
}

.GarbageInformation {
  @include select(PageContent) {
    padding-bottom: 5rem;
  }
}

.InfoSection {
  transition: max-width 400ms ease;
}

.Panel {
  padding: 0 0 2rem 0;

  &:first-child {
    padding-top: 0;
  }

  @include select(Heading_Heading__tiny) {
    color: $color-neutral-grey4;
    font-weight: normal;
    @include GarbagePanel;
  }

  .htmlContent,
  > p {
    @include GarbagePanel;
    margin-bottom: 2rem;
  }

  .htmlContent h4,
  .htmlContent h3 {
    font-size: 1.6rem;
    margin: 0;
  }
  .htmlContent > p {
    &:first-child {
      margin-top: 0;
    }
  }
  .htmlContent,
  > p {
    > a {
      color: inherit;
      text-decoration: none;
    }
  }
}

.Panel {
  .AfvalpuntContentHeading {
    @include GarbagePanel;
    margin-bottom: 2rem;
  }
}

.AddressPanel {
  max-width: $main-boxed-width;
  margin: 0 auto;
  padding-top: 2rem;
  margin-bottom: 2rem;
  border-top: 0.1rem solid $color-neutral-grey2;

  > p {
    font-weight: bold;
    font-size: 1.4em;
  }

  @include mq-tablet() {
    margin: 0 2rem;
  }

  @include mq-phone() {
    margin: 0 1rem;
  }
}

.DistanceToHome {
  font-size: 1.3rem;
  color: $color-neutral-grey4;
  font-weight: normal;
}

.ToggleOtherGarbageCentersButton {
  max-width: 64rem;
  margin: 0 auto;

  @include select(Button_Linkd) {
    cursor: pointer;
    @include select(Button_Icon) {
      transition: transform 100ms ease-in-out;
    }
  }
}

.InfoSectionMap {
  border: 0;
  // hairline that represents the same width as other sections. The map is set at 100% vw, the default border will expand to that with and thus mismatching design.
  &:after {
    position: absolute;
    display: block;
    content: '';
    border-bottom: .1rem solid $color-neutral-grey2;
    width: $main-boxed-width;
    max-width: calc(100% - 2rem);
    left: 50%;
    transform: translateX(-50%);
  }
  &[class*=SectionCollapsible]:not([class*=SectionCollapsible_isCollapsed]) {
    max-width: 100%;
  }
}

.OpeningHours {
   > *:not(:first-child) {
    background-color: $color-neutral-grey1;
    padding: 1rem;
    margin: 0;
    // display: none;
  }

  > ul {
    padding-left: 3rem!important; // sorry...
  }
}

.ContactLink {
  margin-left: -1.8rem;

  @include mq-phone {
    margin-left: 0;
  }
}
