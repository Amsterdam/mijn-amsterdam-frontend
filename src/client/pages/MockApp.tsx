import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
