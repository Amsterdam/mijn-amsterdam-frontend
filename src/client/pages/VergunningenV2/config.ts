export const displayPropsHuidigeVergunningen = {
  identifier: 'Kenmerk',
  title: 'Soort vergunning',
  dateRequest: 'Aangevraagd',
};

export const displayPropsEerdereVergunningen = {
  identifier: 'Kenmerk',
  title: 'Soort vergunning',
  decision: 'Resultaat',
};

export const listPageParamKind = {
  actual: 'huidige-vergunningen-en-ontheffingen',
  historic: 'eerdere-vergunningen-en-ontheffingen',
};

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

export const listPageTitle = {
  [listPageParamKind.actual]: 'Huidige vergunningen en ontheffingen',
  [listPageParamKind.historic]:
    'Eerdere en niet verleende vergunningen en ontheffingen',
};
