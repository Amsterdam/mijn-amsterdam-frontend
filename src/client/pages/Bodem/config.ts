import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
