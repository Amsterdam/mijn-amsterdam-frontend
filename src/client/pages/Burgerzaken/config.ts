import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
