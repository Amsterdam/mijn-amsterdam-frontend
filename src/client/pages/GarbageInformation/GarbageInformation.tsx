import { Link } from '@amsterdam/design-system-react';
import classNames from 'classnames';

import styles from './GarbageInformation.module.scss';
import { AppRoutes } from '../../../universal/config/routes';
import { isError, isLoading } from '../../../universal/helpers/api';
import { getFullAddress, isMokum } from '../../../universal/helpers/brp';
import {
  GarbageFractionCode,
  GarbageFractionInformationTransformed,
} from '../../../universal/types';
import {
  IconAfvalGft,
  IconAfvalGlas,
  IconAfvalGrofAfval,
  IconAfvalPapier,
  IconAfvalRest,
  IconAfvalTextiel,
} from '../../assets/icons/map';
import {
  Button,
  DetailPage,
  ErrorAlert,
  InfoDetail,
  InnerHtml,
  Linkd,
  LinkdInline,
  LoadingContent,
  MaintenanceNotifications,
  PageContent,
  PageHeading,
  ThemaIcon,
} from '../../components';
import { ButtonBody, buttonStyle } from '../../components/Button/Button';
import { InfoDetailProps } from '../../components/InfoDetail/InfoDetail';
import { ExternalUrls } from '../../config/app';
import { ThemaTitles } from '../../config/thema';
import { useAppStateGetter } from '../../hooks/useAppState';
import { useProfileTypeValue } from '../../hooks/useProfileType';
import { useTermReplacement } from '../../hooks/useTermReplacement';

function GarbageInfoDetail({ ...props }: InfoDetailProps) {
  return <InfoDetail {...props} className={styles.GarbageInfoDetail} />;
}

interface InstructionCTAProps {
  fraction: GarbageFractionInformationTransformed;
}

function ButtonLink({
  children,
  href,
  external,
}: {
  children: any;
  href: string;
  external: boolean;
}) {
  const relProp = external ? { rel: 'external noopener noreferrer' } : null;
  return (
    <a
      className={buttonStyle({
        variant: 'secondary',
        className: classNames(styles.ProfileLink, styles.LogoutLink),
      })}
      href={href}
      {...relProp}
    >
      <ButtonBody>{children}</ButtonBody>
    </a>
  );
}

function InstructionCTA({ fraction }: InstructionCTAProps) {
  if (fraction.instructieCTA) {
    return (
      <>
        <p>
          <ButtonLink
            href={fraction.instructieCTA.to}
            external={!fraction.instructieCTA.to.startsWith(AppRoutes.BUURT)}
          >
            {fraction.instructieCTA.title}
          </ButtonLink>
        </p>
        {fraction.instructie ? (
          <>
            <InnerHtml el="p">{fraction.instructie}</InnerHtml>
          </>
        ) : null}
      </>
    );
  }
  return fraction.instructie ? (
    <InnerHtml el="p">{fraction.instructie}</InnerHtml>
  ) : null;
}

interface GarbageFractionIconProps {
  fractionCode: GarbageFractionCode;
}

function GarbageFractionIcon({ fractionCode }: GarbageFractionIconProps) {
  let icon = <IconAfvalRest />;
  switch (fractionCode.toLowerCase()) {
    case 'ga':
      icon = <IconAfvalGrofAfval />;
      break;
    case 'glas':
      icon = <IconAfvalGlas />;
      break;
    case 'papier':
      icon = <IconAfvalPapier />;
      break;
    case 'textiel':
      icon = <IconAfvalTextiel />;
      break;
    case 'gft':
      icon = <IconAfvalGft />;
      break;
    default:
    case 'rest':
      icon = <IconAfvalRest />;
      break;
  }
  return <span className={styles.GarbageFractionIcon}>{icon}</span>;
}

interface GarbageFractionPanelProps {
  fraction: GarbageFractionInformationTransformed;
}

function GarbageFractionPanel({ fraction }: GarbageFractionPanelProps) {
  return (
    <article className={styles.GarbageFractionPanel}>
      <h3 className={styles.GarbageFractionPanelTitle}>
        <GarbageFractionIcon fractionCode={fraction.fractieCode} />
        {fraction.titel}
      </h3>
      {!!fraction.kalendermelding && (
        <p className={styles.GarbageFractionPanelHighlight}>
          <InnerHtml>{fraction.kalendermelding}</InnerHtml>
        </p>
      )}
      {(fraction.instructieCTA || fraction.instructie) && (
        <dl>
          <dt>Hoe</dt>
          <dd>
            <InstructionCTA fraction={fraction} />
          </dd>
        </dl>
      )}
      {!!fraction.ophaaldagen && (
        <dl>
          <dt>Ophaaldag</dt>
          <dd>{fraction.ophaaldagen}</dd>
        </dl>
      )}
      {!!fraction.buitenzetten && (
        <dl>
          <dt>Buitenzetten</dt>
          <dd>{fraction.buitenzetten}</dd>
        </dl>
      )}
      {!!fraction.waar && (
        <dl>
          <dt>Waar</dt>
          <dd>
            {typeof fraction.waar === 'object' ? (
              <LinkdInline
                href={fraction.waar.to}
                external={!fraction.waar.to.startsWith(AppRoutes.BUURT)}
              >
                {fraction.waar.title}
              </LinkdInline>
            ) : (
              fraction.waar
            )}
          </dd>
        </dl>
      )}
      {!!fraction.opmerking && (
        <div className={styles.GarbageFractionPanelOpmerking}>
          <InnerHtml>{fraction.opmerking}</InnerHtml>
        </div>
      )}
    </article>
  );
}

interface GarbageFractionPanelsProps {
  fractions: GarbageFractionInformationTransformed[];
}

const fractions1 = ['rest', 'ga', 'papier'];
const fractions2 = ['gft', 'glas', 'textiel'];

function GarbageFractionPanels({ fractions }: GarbageFractionPanelsProps) {
  const fractionsByCode = Object.fromEntries(
    fractions.map((fraction) => [fraction.fractieCode.toLowerCase(), fraction])
  );

  return (
    <div className={styles.GarbageFractionPanels}>
      <div className={styles.GarbageFractionPanelsColumn}>
        {fractions1
          .filter((fractionCode) => fractionCode in fractionsByCode)
          .map((fractionCode) => (
            <GarbageFractionPanel
              key={fractionCode}
              fraction={fractionsByCode[fractionCode]}
            />
          ))}
      </div>
      <div className={styles.GarbageFractionPanelsColumn}>
        {fractions2
          .filter((fractionCode) => fractionCode in fractionsByCode)
          .map((fractionCode) => (
            <GarbageFractionPanel
              key={fractionCode}
              fraction={fractionsByCode[fractionCode]}
            />
          ))}
      </div>
    </div>
  );
}

export default function GarbageInformation() {
  const { AFVAL, AFVALPUNTEN, MY_LOCATION, BRP, KVK } = useAppStateGetter();
  const profileType = useProfileTypeValue();
  const termReplace = useTermReplacement();
  const address = MY_LOCATION.content?.[0]?.address;
  const inMokum =
    profileType === 'private' ? isMokum(BRP.content) : isMokum(KVK.content);
  const isWeesp = address?.woonplaatsNaam === 'Weesp';
  const isApiReady = !isLoading(MY_LOCATION) && !isLoading(AFVAL);

  const heeftGeenWoonfunctie = AFVAL.content?.some(
    (fractionData) => !fractionData.gebruiksdoelWoonfunctie
  );
  const commercialLocation = MY_LOCATION.content?.find(
    (location) => location?.profileType === 'commercial'
  );
  const privateLocation = MY_LOCATION.content?.find(
    (location) => location?.profileType === 'private'
  );

  const privateIsCommercial =
    commercialLocation?.bagNummeraanduidingId ===
    privateLocation?.bagNummeraanduidingId;

  return (
    <DetailPage>
      <PageHeading
        backLink={{
          to: AppRoutes.HOME,
          title: 'Home',
        }}
        isLoading={false}
        icon={<ThemaIcon />}
      >
        {termReplace(ThemaTitles.AFVAL)}
      </PageHeading>
      <PageContent>
        {profileType === 'private' && (
          <>
            <p>Dit zijn de afvalregels voor uw adres.</p>
            {!!commercialLocation && !privateIsCommercial && (
              <ErrorAlert title="Bedrijfsafval informatie" severity="info">
                Let op deze regels gaan over uw woonadres. Lees hier{' '}
                <LinkdInline
                  href={ExternalUrls.AFVAL_COMMERCIAL}
                  external={true}
                >
                  regels over bedrijfsafval in Amsterdam
                </LinkdInline>
                .
              </ErrorAlert>
            )}
            <p>
              <Linkd href={ExternalUrls.AFVAL} external={true}>
                Meer informatie over regels voor afval en hergebruik
              </Linkd>
              <br />
              <Linkd href={ExternalUrls.AFVAL_MELDING} external={true}>
                Doe een melding als afval is blijven liggen
              </Linkd>
            </p>
          </>
        )}
        {profileType !== 'private' && (
          <>
            <p>
              De afvalregels hieronder gelden alleen als u maximaal 9
              vuilniszakken afval per week heeft en reinigingsrecht betaalt. Of
              als u ondernemer bent in de 9 straatjes of Sluisbuurt en
              afvalstoffenheffing betaalt. Anders moet u een
              bedrijfsafvalcontract afsluiten bij een{' '}
              <Link
                rel="noopener noreferrer"
                href="https://www.afvalgids.nl/afval/inzamelaar/"
              >
                erkende afvalinzamelaar
              </Link>{' '}
              of met{' '}
              <Link
                rel="noopener noreferrer"
                href="https://www.amsterdam.nl/afval-hergebruik/bedrijfsafval/bedrijfsafval-laten-ophalen/"
              >
                de gemeente
              </Link>
              .
            </p>
            <p>
              <Linkd href={ExternalUrls.AFVAL_COMMERCIAL} external={true}>
                Regels bedrijfsafval in Amsterdam
              </Linkd>
              <br />
              <Linkd href={ExternalUrls.AFVAL_MELDING} external={true}>
                Doe een melding als afval is blijven liggen
              </Linkd>
            </p>
          </>
        )}
        <MaintenanceNotifications page="afval" />
        {isError(AFVAL) && (
          <ErrorAlert>
            We kunnen op dit moment niet alle gegevens tonen
          </ErrorAlert>
        )}

        <GarbageInfoDetail
          label="Uw adres"
          valueWrapperElement="div"
          value={
            <>
              <p className={styles.AdresWeergave}>
                {MY_LOCATION.content?.[0]?.address ? (
                  getFullAddress(MY_LOCATION.content?.[0].address)
                ) : isLoading(MY_LOCATION) ? (
                  <LoadingContent barConfig={[['20rem', '3rem', '0']]} />
                ) : (
                  'Onbekend adres'
                )}
              </p>
              {/* NOTE: Edge case: Een (niet zakelijke) burger kan ingeschreven zijn op een pand zonder woonfunctie. */}
              {heeftGeenWoonfunctie && (
                <p className={styles.WoonFunctieWaarschuwing}>
                  <strong>Dit is geen woonadres.</strong> Klopt dit niet?{' '}
                  <LinkdInline
                    external
                    href="https://formulier.amsterdam.nl/thema/afval-grondstoffen/klopt-afvalwijzer/Reactie"
                  >
                    Geef het door
                  </LinkdInline>
                  .
                </p>
              )}
            </>
          }
        />
        {isApiReady && isWeesp && (
          <p className={styles.WeespWarning}>
            In Weesp haalt de GAD het afval op. Kijk op de{' '}
            <a href="https://www.gad.nl/">website van de GAD</a> hoe dat werkt.
          </p>
        )}
        {isApiReady && inMokum && (
          <>
            {AFVAL.status === 'OK' && !!AFVAL.content?.length && (
              <GarbageFractionPanels fractions={AFVAL.content} />
            )}
            <p>
              <Button
                className={styles.ContactLink}
                onClick={() =>
                  (window.location.href = ExternalUrls.AFVAL_MELDING_FORMULIER)
                }
              >
                Klopt de informatie niet? Geef het door
              </Button>
            </p>
            <h3>Afvalcontainers in de buurt</h3>
            <p>
              <Linkd
                href={`${AppRoutes.BUURT}?datasetIds=["afvalcontainers"]&zoom=14`}
              >
                Kaart met containers in de buurt
              </Linkd>
            </p>
            <h3>Adressen recyclepunten</h3>
            <ul className={styles.UnstyledList}>
              {AFVALPUNTEN.content?.centers.map((item, index) => (
                <li key={item.title}>
                  <Linkd href={item.website} external={true}>
                    {item.title}
                    {item.distance !== 0 && (
                      <span className={styles.DistanceToHome}>
                        +/-{item.distance}KM
                      </span>
                    )}
                  </Linkd>
                </li>
              ))}
            </ul>
          </>
        )}
      </PageContent>
    </DetailPage>
  );
}
