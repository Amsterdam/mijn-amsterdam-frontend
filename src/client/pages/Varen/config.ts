import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { VarenFrontend } from '../../../server/services/varen/config-and-types';

export const displayPropsAanvragen = {
  detailLinkComponent: '',
  title: 'Omschrijving',
  dateRequestFormatted: 'Aangevraagd',
  status: 'Status',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (vergunning: VarenFrontend) => !vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (vergunning: VarenFrontend) => vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
};
