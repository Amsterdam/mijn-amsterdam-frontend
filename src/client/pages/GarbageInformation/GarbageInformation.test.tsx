import { render } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { AppRoutes } from '../../../universal/config';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import GarbageInformation from './GarbageInformation';

//const { BRP, AFVAL, AFVALPUNTEN, HOME } = useAppStateGetter();

const restafval = {
  title: 'Gooi zakken in container',
  aanbiedwijze: 'In container',
  stadsdeel: 'Zuid-Oost',
  type: 'huisvuil',
  buitenZetten: '',
  ophaaldag: '',
  opmerking: 'Hou het netjes!',
};
const grofvuil = {
  title: 'Zetje afval aan de weg',
  aanbiedwijze: 'Aan de weg',
  stadsdeel: 'Zuid-Oost',
  type: 'grofvuil',
  buitenZetten: 'vanaf Woensdag 20:00 tot Donderdag 07:00',
  ophaaldag: 'Donderdag',
  opmerking: 'Hou het netjes!',
};

const afvalpunt = {
  title: 'De afvalpuntenweg',
  latlng: { latng: { lat: 5, lng: 40 } },
  url: 'http://example.org/afvalpunt',
  address: 'afvalpuntenweg 23, 1067 BA',
  phone: '123123123',
  email: 'afval@punt.com',
  distance: 34,
  openingHours: 'van maandag t/m vrijdag 09:00 16:30',
};

const testState: any = {
  BRP: {
    status: 'OK',
    content: {
      adres: {
        straatnaam: 'Teststraat',
        huisnummer: '24',
        huisnummertoevoeging: 'x',
        huisletter: null,
      },
    },
  },
  AFVAL: { status: 'OK', content: [restafval, grofvuil] },
  AFVALPUNTEN: { status: 'OK', content: { centers: [afvalpunt] } },
  HOME: { status: 'OK', content: { latlng: { lat: 'x', lng: 'y' } } },
};

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<GarbageInformation />', () => {
  const routeEntry = generatePath(AppRoutes.AFVAL);
  const routePath = AppRoutes.AFVAL;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={GarbageInformation}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
