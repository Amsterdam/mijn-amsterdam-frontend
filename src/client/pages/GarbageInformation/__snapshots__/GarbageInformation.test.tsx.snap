// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GarbageInformation /> Matches the Full Page snapshot 1`] = `"<main class=\\"Page DetailPage GarbageInformation\\"><header class=\\"PageHeading hasIcon\\"><span aria-hidden=\\"true\\" class=\\"Icon\\"><svg aria-label=\\"AFVAL\\" fill=\\"black\\" class=\\"ChapterIcon\\">Huisvuilkalender.svg</svg></span><div class=\\"HeadingInner\\"><h2 class=\\"Heading Heading__large\\">Afval</h2></div></header><div class=\\"PageContent\\"><p>Bekijk waar u uw afval kwijt kunt en hoe u uw afval kunt scheiden.</p><p><a rel=\\"external noopener noreferrer\\" href=\\"https://www.amsterdam.nl/afval/\\" class=\\"Button Button__plain Button__lean Linkd\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" fill=\\"#000\\" class=\\"Icon Icon__left\\">Chevron-Right.svg</svg>De regels voor afval en hergebruik</a></p></div><div class=\\"Panel Panel AddressPanel\\"><h3 class=\\"Heading Heading__tiny\\">Uw adres</h3><p>Teststraat 24  x</p></div><section class=\\"SectionCollapsible InfoSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\"><button aria-expanded=\\"false\\" class=\\"TitleToggle\\"><svg aria-hidden=\\"true\\" class=\\"CaretIcon\\">Chevron-Right.svg</svg> Zetje afval aan de weg</button></h3><div class=\\"Panel\\" style=\\"height: 0px;\\" aria-hidden=\\"true\\"><div class=\\"PanelInner\\"><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Hoe</h3><p>Aan de weg</p></div><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Buiten zetten</h3><p>vanaf Woensdag 20:00 tot Donderdag 07:00</p></div><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Ophaaldag</h3><p>Donderdag</p></div><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Opmerking</h3><p>Hou het netjes!</p></div></div></div></section><section class=\\"SectionCollapsible InfoSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\"><button aria-expanded=\\"false\\" class=\\"TitleToggle\\"><svg aria-hidden=\\"true\\" class=\\"CaretIcon\\">Chevron-Right.svg</svg> Gooi zakken in container</button></h3><div class=\\"Panel\\" style=\\"height: 0px;\\" aria-hidden=\\"true\\"><div class=\\"PanelInner\\"><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Hoe</h3><p>In container</p></div><div class=\\"Panel Panel\\"><h3 class=\\"Heading Heading__tiny\\">Opmerking</h3><p>Hou het netjes!</p></div></div></div></section><section class=\\"SectionCollapsible InfoSection InfoSectionMap\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\"><button aria-expanded=\\"false\\" class=\\"TitleToggle\\"><svg aria-hidden=\\"true\\" class=\\"CaretIcon\\">Chevron-Right.svg</svg> Afvalcontainers in de buurt</button></h3><div class=\\"Panel\\" style=\\"height: 0px;\\" aria-hidden=\\"true\\"><div class=\\"PanelInner\\"><div class=\\"MyAreaMapContainer GarbageContainerMap\\"><div class=\\"MyAreaLoader\\"><span><svg aria-hidden=\\"true\\">home.svg</svg>Uw adres wordt opgezocht...</span></div></div></div></div></section><section class=\\"SectionCollapsible InfoSection InfoSectionGarbagePoints\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\"><button aria-expanded=\\"false\\" class=\\"TitleToggle\\"><svg aria-hidden=\\"true\\" class=\\"CaretIcon\\">Chevron-Right.svg</svg> Afvalpunten</button></h3><div class=\\"Panel\\" style=\\"height: 0px;\\" aria-hidden=\\"true\\"><div class=\\"PanelInner\\"><div class=\\"Panel Panel AfvalPunten\\"><h3 class=\\"Heading Heading__medium AfvalpuntContentHeading\\">De afvalpuntenweg <span class=\\"DistanceToHome\\">+/-34KM</span></h3><h3 class=\\"Heading Heading__tiny\\">Adres</h3><p>afvalpuntenweg 23, 1067 BA</p><h3 class=\\"Heading Heading__tiny\\">Telefoon</h3><p><a href=\\"tel:123123123\\">123123123</a></p><h3 class=\\"Heading Heading__tiny\\">E-mail</h3><p><a href=\\"mailto:afval@punt.com\\">afval@punt.com</a></p><h3 class=\\"Heading Heading__tiny\\">Openingstijden</h3><div class=\\"htmlContent OpeningHours\\">van maandag t/m vrijdag 09:00 16:30</div></div></div></div></section><div class=\\"PageContent\\"><p><a rel=\\"external noopener noreferrer\\" href=\\"https://formulier.amsterdam.nl/mail/afval/afvalwijzer/\\" class=\\"Button Button__plain Button__lean Linkd ContactLink\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" fill=\\"#000\\" class=\\"Icon Icon__left\\">Chevron-Right.svg</svg>Kloppen de dagen, tijden of adressen niet? Geef het aan ons door.</a></p></div></main>"`;
